test/unittests/test_intx_api.cpp

```diff
@@ -259,6 +259,49 @@ TYPED_TEST(uint_api, comparison)
     EXPECT_TRUE(d >= x);
 }
 
+template <typename T>
+struct Wrapper
+{
+    T v;
+
+    friend auto operator<=>(Wrapper, Wrapper) = default;
+};
+
+TYPED_TEST(uint_api, spaceship_operator_with_wrapper)
+{
+    auto a = Wrapper<TypeParam>{TypeParam{1}};
+    auto b = Wrapper<TypeParam>{TypeParam{2}};
+    auto c = Wrapper<TypeParam>{TypeParam{3}};
+
+    EXPECT_EQ(a <=> b, std::strong_ordering::less);
+    EXPECT_EQ(a <=> a, std::strong_ordering::equal);
+    EXPECT_EQ(c <=> b, std::strong_ordering::greater);
+
+    EXPECT_TRUE(a < b);
+    EXPECT_TRUE(c > a);
+    EXPECT_TRUE(a >= a);
+    EXPECT_TRUE(b <= c);
+    EXPECT_TRUE(b == b);
+    EXPECT_TRUE(a != c);
+}
+
+TYPED_TEST(uint_api, spaceship_operator)
+{
+    auto a = TypeParam{1};
```

Comment:
Please use syntax without `auto`. This is now my stylistic preference, nothing more.

```suggestion
    TypeParam a{1};
```